ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -39,13 +61,76 @@ const PurpleLabel = ({style = {}, children}) => {
   </label>;
 };
 
+const workspaceInfoField = (labelText: string, divContents: React.ReactFragment) => {
+  return <FlexRow style={styles.infoRow}>
+    <PurpleLabel style={styles.infoLabel}>{labelText}</PurpleLabel>
+    <div style={styles.infoValue}>{divContents}</div>
+  </FlexRow>;
+};
+
+const FileDetailsTable = (props: {data: Array<FileDetail>, bucket: string}) => {
+  const {data, bucket} = props;
+
+  interface TableEntry {
+    name: string;
+    size: string;
+    location: string;
+  }
+
+  const getLocation = (file: FileDetail): string => {
+    const prefix = `${bucket}/`;
+    const suffix = `/${file.name}`;
+    return file.path.replace(prefix, '').replace(suffix, '');
```

Comment:
Current approach incorrectly handles this input:

```
parseLocation({... 'gs://bkt/foo/bar/foo'}) -> 'gs://bkt/bar/foo'
```

Probably `substring()` is best here; regexp is a bit dicey since the prefix/suffix are ultimately user input.

Reply:
Sorry - not quite following here

Reply:
`replace` replaces the first instance, so if the filename is duplicate in the path this approach fails

Reply:
ah thanks